lib/src/ovis_event/ovis_event.c

```diff
@@ -79,6 +79,8 @@ void ovis_scheduler_ref_get(ovis_scheduler_t m)
 static inline
 void ovis_scheduler_ref_put(ovis_scheduler_t m)
 {
+	if (!m)
+		return;
 	pthread_mutex_lock(&m->mutex);
 	assert(m->refcount > 0);
```

Comment:
I know this bug is unrelated to your fix, but this assert belongs before the pthread_mutex_lock() call.